### ISSUE TITLE
vimwiki_autowriteall failing to write when navigating between pages

### ISSUE THREAD

**Issue by novelgazer** [vimwiki/vimwiki#1448]
Prior to submitting a new issue make sure to complete these steps:

- [X] Include the VimWiki settings from your `.vimrc`
  - let g:vimwiki_list=[{'path': '/home/sean/Vimwiki/', 'syntax': 'markdown', 'ext': '.md'}]

- [X] Include the syntax you are using (default / Markdown / MediaWiki) 
  - Markdown

- [X] Provide a detailed description of the problem including **steps to reproduce the issue**.
  - Open vimwiki with Leader ww
  - Open any page
  - Modify the page
  - Backspace to return to previous page shows error: Vimwiki: Warning: Can't leave the current buffer, because it is modified. Hint: Take a look at':h g:vimwiki_autowriteall' to see how to save automatically.

- [X] Include the output of `:VimwikiShowVersion`.

Version: 2024.01.24
Os: Linux
Vim: 901
Branch: dev
Revision: 7279261
Date: 2024-10-13 22:07:48 -0600

Vim version is 9.1.825

**Comment by claman** [comment]
Not able to reproduce this on revision 705ad1e

Updated to 7279261 and also not seeing it; tested on two different wikis.

**Comment by novelgazer** [comment]
Thanks for testing, @claman Could it be related to the version of Vim?

**Comment by novelgazer** [comment]
If it's of any interest, when I run :let I see the following, which suggests autowriteall is on and should be working.  Behavior was as expected until some very recent change.  (I can't pinpoint exactly when.)

> vimwiki_global_vars   {'conceallevel': 2, 'autowriteall': 1

 (etc., etc.)

**Comment by novelgazer** [comment]
Further testing shows that this only affects wiki files that I've set up in a weird way for myself to facilitate syncing across devices: they exist as .txt files in a different folder (dropbox) and within the wiki they are simlinks to the original files with .md extensions.  Files not in this state autowrite without any problems.  Not sure why this behavior suddenly broke after at least a year of working as expected.  To resolve it, I defined a second wiki in the dropbox folder using .txt extensions, and that seems to have worked.

.vimrc now has this line:

> let g:vimwiki_list=[{'path': '/home/sean/Vimwiki/', 'syntax': 'markdown', 'ext': '.md'},{'path': '/home/sean/Dropbox/', 'syntax': 'markdown', 'ext': '.txt'}]
